Yoast/wordpress-seo
# Remove the folders that are already in the vendor_prefixed folder from the vendor folder

## Summary

This PR can be summarized in the following changelog entry:

* [not user-facing] Removes the folders that are already in the vendor_prefixed folder from the vendor folder

## Relevant technical choices:
* There is no reason to have dependencies both in the vendor_prefixed folder as well as in the vendor folder. 
* This PR introduces a `'shell:remove-prefixed-sources'` command within the `grunt artifact` alias that removes a hard-coded list of dependencies from the composer.json and composer.lock, because they are no longer dependencies.
* This PR also introduces a `'restore-environment-after-artifact'` command that is doing a `composer install`, and resets the composer.lock and composer.json (that were altered by the above command) to make sure your local installation is still functioning after running `grunt artifact` and the changed composer.json and composer.lock won't be committed.

## Test instructions

This PR can be tested by following these steps:
* Run `grunt artifact` on release/10.0 and run `grunt artifact` on this branch. Compare the vendor folders of the artifact folder, and see that the artifact folder from this branch doesn't contain:
```
				"vendor/guzzlehttp",
				"vendor/j4mie",
				"vendor/league",
				"vendor/psr",
				"vendor/ruckusing",
				"vendor/bin/ruckus.php",
```
* Create a zip from this branch using the create_beta script.
* Install the zip, click around a bit, and see that nothing is broken. Oauth has not been implemented completely yet in 10.0 (from which this is branched, which means we cannot really test yet of the oauth-related stuff in the plugin is affected by this PR)
* Do the same for Premium. I've created a `remove-folders-from-vendor-folder-2` branch on premium, which has this `wordpress-seo` branch merged into it.
* Because the issue I'm fixing with this PR is that the beta zip was >5MB, I've built a new zip with these changes. The resulting zip is 4.9 MB 🎉. See the issue for the zip. 

## UI changes

## Quality assurance


Fixes Yoast/wordpress-seo-premium/issues/2263
